lib/reek/smell_detectors/nested_iterators.rb

```diff
@@ -128,8 +128,15 @@ def max_allowed_nesting
 
       # @quality :reek:FeatureEnvy
       def ignored_iterator?(exp)
-        ignore_iterators.any? { |pattern| /#{pattern}/ =~ exp.call.name } ||
-          exp.without_block_arguments?
+        ignore_iterators.any? do |pattern|
+          call = exp.call
+          called_name = call.name
+          /#{pattern}/ =~ if call.respond_to? :receiver
+                            "#{call.receiver&.name}.#{called_name}"
+                          else
+                            called_name
+                          end
```

Comment:
The following seems to work:

```ruby
/#{pattern}/ =~ call.format_to_ruby
```

Reply:
Interesting. I never heard of that method. Will edit the PR

Reply:
It's a method defined on AST nodes by Reek itself.

Reply:
Applied changes